.github/workflows/test_e2e.yml

```
@@ -1,11 +1,20 @@
 name: Test e2e
 
-on: [push]
+on:
+  pull_request:
+    branches:
+      - main
 
 jobs:
+
   end-to-end-tests:
     runs-on: ubuntu-latest
     steps:
+      - name: ready to execute e2e test
+        if: "!contains( github.event.pull_request.labels.*.name , 'run e2e tests' )"
+        run: |
+          echo "::error add label 'ensure e2e tests' to execute e2e tests, after adding the label execute this job again"
+          exit 1
```

I would suggest that is better only to run the job with his steps if the label is set
```suggestion
  end-to-end-tests:
    if: "contains( github.event.pull_request.labels.*.name , 'run e2e tests' )"
    runs-on: ubuntu-latest
    steps:
```